cyral/resource_cyral_repository_test.go

```diff
@@ -11,146 +12,343 @@ const (
+		Type:   "mongodb",
+		Labels: []string{"rds", "us-east-2"},
+		RepoNodes: []*RepoNode{
+			{
+				Host: "mongo.local",
+				Port: 3333,
+			},
+		},
+	}
 
-var emptyPropertiesRepoConfig RepoData = RepoData{
-	Name:       accTestName(repositoryResourceName, "repo-empty-properties"),
-	Host:       "mongo-cluster.local",
-	Port:       27017,
-	RepoType:   "mongodb",
-	Properties: &RepositoryProperties{},
-}
+	updatedRepoConfig = RepoInfo{
+		Name:   accTestName(repositoryResourceName, "repo-updated"),
+		Type:   "mongodb",
+		Labels: []string{"rds", "us-east-1"},
+		RepoNodes: []*RepoNode{
+			{
+				Host: "mongo.local",
+				Port: 3334,
+			},
+		},
+	}
 
-var replicaSetRepoConfig RepoData = RepoData{
-	Name:                accTestName(repositoryResourceName, "repo-replica-set"),
-	Host:                "mongo-cluster.local",
-	Port:                27017,
-	RepoType:            "mongodb",
-	MaxAllowedListeners: 2,
-	Properties: &RepositoryProperties{
-		MongoDBReplicaSetName: "replica-set-1",
-		MongoDBServerType:     mongodbReplicaSetServerType,
-	},
-}
+	emptyConnDrainingConfig = RepoInfo{
+		Name: accTestName(repositoryResourceName, "repo-empty-conn-draining"),
+		Type: "mongodb",
+		ConnParams: &ConnParams{
+			ConnDraining: &ConnDraining{},
+		},
+		RepoNodes: []*RepoNode{
+			{
+				Host: "mongo-cluster.local",
+				Port: 27017,
+			},
+		},
+	}
 
-func TestAccRepositoryResource(t *testing.T) {
-	testConfig, testFunc := setupRepositoryTest(
-		initialRepoConfig, "update_test")
-	testUpdateConfig, testUpdateFunc := setupRepositoryTest(
-		updatedRepoConfig, "update_test")
-	testEmptyPropertiesConfig, testEmptyPropertiesFunc := setupRepositoryTest(
-		emptyPropertiesRepoConfig, "empty_properties_test")
-	testReplicaSetConfig, testReplicaSetFunc := setupRepositoryTest(
-		replicaSetRepoConfig, "replica_config_test")
+	connDrainingConfig = RepoInfo{
+		Name: accTestName(repositoryResourceName, "repo-conn-draining"),
+		Type: "mongodb",
+		ConnParams: &ConnParams{
+			ConnDraining: &ConnDraining{
+				Auto:     true,
+				WaitTime: 20,
+			},
+		},
+		RepoNodes: []*RepoNode{
+			{
+				Host: "mongo-cluster.local",
+				Port: 27017,
+			},
+		},
+	}
 
-	// Should use name of the last resource created.
-	importTestResourceName := "cyral_repository.replica_config_test"
+	accessGatewayEmptyConfig = RepoInfo{
+		Name: accTestName(repositoryResourceName, "repo-with-access-gateway-empty"),
+		Type: "mongodb",
+		ConnParams: &ConnParams{
+			ConnDraining: &ConnDraining{
+				Auto:     true,
+				WaitTime: 20,
+			},
+		},
+		RepoNodes: []*RepoNode{
+			{
+				Name: "node1",
+				Host: "mongo.local.node1",
+				Port: 27017,
+			},
+			{
+				Name: "node2",
+				Host: "mongo.local.node2",
+				Port: 27017,
+			},
+		},
+		PreferredAccessGwBinding: &BindingKey{},
+	}
 
-	resource.ParallelTest(t, resource.TestCase{
-		ProviderFactories: providerFactories,
-		Steps: []resource.TestStep{
+	// accessGatewayConfig = RepoInfo{
```

Comment:
The breaking changes to repos/bindings/listeners include a change that does not allow the API user to set the access gateway sidecar, without a valid sidecarID and bindingID. Since the breaking changes to cyral_repository_binding are not yet merged in (and also depend on this code), I have left that test commented out for now. Once this PR merged into `feature/multiplexing`, I will rebase the other binding resource changes of off `feature/multiplexing`, and uncomment the tests / edit that test step.